Sync/SyncStateMachine.swift

```diff
@@ -145,6 +165,19 @@ open class SyncStateMachine {
             }
             b.hashedUID = token.hashedFxAUID
 
+            var enginesEnablements: [String: Bool] = [:]
+            LocalEngines.forEach { engine in
+                let stateChangedPref = "engine.\(engine).enabledStateChanged"
```

Comment:
Note: If the settings views wrote directly to Scratchpad.PrefEnginesEnablements, we wouldn't have to do any of this.
However I don't think it's a good idea for external actors to write directly into scratchpad prefs.

Reply:
This is why I suggested taking an optional engine enablement input as part of advancing — it's "go to `Ready`, and if you want to know what the user picked, here it is". 

You can remove the need to access prefs altogether by doing two things:

- Providing a structured enablement input here (e.g., as part of `SyncAuthData`
- Removing the value from prefs as a consequence of `toReady` succeeding, rather than doing it inside the state machine.